diff --git a/node_modules/react-native-screens/ios/RNSScreen.m b/node_modules/react-native-screens/ios/RNSScreen.m
index 1de4182..df8198a 100644
--- a/node_modules/react-native-screens/ios/RNSScreen.m
+++ b/node_modules/react-native-screens/ios/RNSScreen.m
@@ -305,6 +305,20 @@ - (void)viewDidLayoutSubviews
   }
 }
 
+
+- (void)viewWillTransitionToSize:(CGSize)size withTransitionCoordinator:(id<UIViewControllerTransitionCoordinator>)coordinator
+{
+  double delayInSeconds = 0.1;
+  dispatch_time_t popTime = dispatch_time(DISPATCH_TIME_NOW, (int64_t)(delayInSeconds * NSEC_PER_SEC));
+  dispatch_after(popTime, dispatch_get_main_queue(), ^(void){
+    [self.view.reactSubviews enumerateObjectsUsingBlock:^(UIView * _Nonnull view, NSUInteger idx, BOOL * _Nonnull stop) {
+      if ([view isKindOfClass:[RNSScreenStackHeaderConfig class]]) {
+        [RNSScreenStackHeaderConfig willShowViewController:self animated:NO withConfig:(RNSScreenStackHeaderConfig *)view];
+      }
+    }];
+  });
+}
+
 - (id)findFirstResponder:(UIView*)parent
 {
   if (parent.isFirstResponder) {
